### PR TITLE
allow concurrent uploads

### PR DESCRIPTION
**Pull request by dmfenton**
This PR takes uploading to S3 out of the export worker "job." Once a file is written, the job is considered over and the worker can process another. This is accomplished by calling `done` then handing off writing to S3 to another function that is outside the job. The result is that we can process more exports over time because we are not waiting for uploads to finish before starting new exports. We also do not wait for the "latest" file to upload which is really not part of job anyhow.

TL;DR: We don't wait for uploads to finish before starting new export jobs. This gives us significantly higher throughput from each worker process.

cc @jwieringa

### CODE REVIEW

**Comment by jwieringa** [comment]
Higher throughput sounds great. Couple of questions:

1) How could we use the job system to re-run or know if there were any failed uploads? (understanding that it is important the processes finish on a box where the file was written)
2) My knowledge of Node is not great; however, in other places I've seen Node configured to create as many "job processes" until the CPU limit is reached. Is that a possibility here?

Would be great to do some benchmarks on the disks with this change.

**Comment by dmfenton** [comment]
1. At this time, each upload is broken into multiple parts -> each of those is retried on failure. Each upload itself is retried once. Retrying just the upload via an API command is out of the scope of this PR, though it would definitely be nice.

2. That is possible, but it will require a switch to node-resque which is well beyond the scope of this PR.

3. I've been running stress tests in our development environment and everything has performed exceptionally well. The S3 managed uploader and the switch to copying vs rewriting the latest files means the box has to do a lot less work. Also, given that we still only process one file conversion at a time, we're not doing any more writes do disk than before. We have the potential to do more disk reads simeltaneously, but in practice S3 uploading happens so fast that we are doing less.
